presto-main/src/main/java/com/facebook/presto/server/security/OpenLdapBinder.java

```diff
@@ -13,32 +13,60 @@
  */
 package com.facebook.presto.server.security;
 
+import com.google.common.base.Throwables;
+import io.airlift.log.Logger;
+
 import javax.inject.Inject;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
 import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import java.util.Optional;
 
+import static com.google.common.base.Preconditions.checkState;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class OpenLdapBinder
         implements LdapBinder
 {
-    private final String baseDistinguishedName;
+    private static final Logger LOG = Logger.get(ActiveDirectoryBinder.class);
+    private final String userObjectClass;
+    private final Optional<String> baseDistinguishedName;
```

Comment:
why is this optional? baseDistinguishedName is a mandatory parameter for password verification